src/app/AddRESTModal.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import {RefreshStudentCallbackType} from "./TranscriptApp";
+import {
+    Button, FormControl, FormLabel, Input,
+    Modal, ModalBody,
+    ModalCloseButton,
+    ModalContent, ModalFooter,
+    ModalHeader,
+    ModalOverlay, NumberDecrementStepper, NumberIncrementStepper, NumberInput, NumberInputField, NumberInputStepper,
+    useDisclosure,
+    useToast
+} from "@chakra-ui/react";
+import {remoteTranscriptManager} from "../client/client";
+import { updateBaseURL } from "../client/remoteService"
+
+export const AddRESTModal: React.FunctionComponent<{ appendTranscript: (newID: number)=>Promise<void> }> = ({appendTranscript}) => {
```

Comment:
Some comments on what the Modal is expected to do could help the reader understand better